src/ln/channelmanager.rs

```diff
@@ -90,15 +91,15 @@ mod channel_held_info {
 	}
 
 	/// Tracks the inbound corresponding to an outbound HTLC
-	#[derive(Clone)]
+	#[derive(Clone, PartialEq)]
 	pub struct HTLCPreviousHopData {
-		pub(super) short_channel_id: u64,
-		pub(super) htlc_id: u64,
-		pub(super) incoming_packet_shared_secret: [u8; 32],
+		pub(crate) short_channel_id: u64,
+		pub(crate) htlc_id: u64,
```

Comment:
You dont need to expose htlc_id/incoming_packet_shared_secret - HTLCSource and HTLCPreviousHopData implement Writeable/Readable directly, so you also dont need to re-implement their serialization/deserialization manually in ChannelMonitor::write/read.

Reply:
Ah damn has written superfluous stuff one more time! Corrected